test/functional/sqlsrv/TC43_FetchData.phpt

```diff
@@ -91,16 +101,29 @@ function checkData($col, $actual, $expected)
     return ($success);
 }
 
-if (! isWindows()) {
-    setUTF8Data(true);
+if ( !isWindows() ) {
+    setlocale(LC_ALL, "en_US.ISO-8859-1");
 }
+
+// test ansi only if windows or non-UTF8 locales are supported (ODBC 17 and above)
+if ( isWindows() || isLocaleSupported() ) {
+    try {
+        setUTF8Data(false);
+        fetchFields();
+    } catch (Exception $e) {
+        echo $e->getMessage();
+    }
+}
+
+// test utf8
 try {
+    setUTF8Data(true);
     fetchFields();
 } catch (Exception $e) {
     echo $e->getMessage();
 }
-setUTF8Data(false);
 
 ?>
 --EXPECT--
 Test "Fetch - Field Data" completed successfully.
+Test "Fetch - Field Data" completed successfully.
```

Comment:
put endtest() at the end of this test so you won't see two of them in the exepcted output
for the case in which you need to skip ansi test you won't invoke this endtest() anyway